arangod/GeneralServer/SslServerFeature.h

```diff
@@ -59,17 +59,33 @@ class SslServerFeature : public application_features::ApplicationFeature {
   void unprepare() override final;
   virtual void verifySslOptions();
 
-  virtual std::shared_ptr<asio_ns::ssl::context> createSslContext();
+  std::unique_ptr<asio_ns::ssl::context> createSslContextInternal(std::string keyfileName,
+                                                                  std::string& content);
+
+  virtual std::shared_ptr<std::vector<std::unique_ptr<asio_ns::ssl::context>>> createSslContext();
+  size_t chooseSslContext(std::string const& serverName);
 
   // Dump all SSL related data into a builder, private keys
   // are hashed.
   virtual Result dumpTLSData(VPackBuilder& builder) const;
 
  protected:
+
+  struct SNIEntry {
+    std::string serverName;      // empty for default
+    std::string keyfileName;     // name of key file
+    std::string keyfileContent;  // content of key file
+    SNIEntry(std::string name, std::string keyfileName)
+      : serverName(name), keyfileName(keyfileName) {}
+  };
+
   std::string _cafile;
   std::string _cafileContent;  // the actual cert file
-  std::string _keyfile;
-  std::string _keyfileContent; // the actual keyfile
+  std::string _keyfile;        // name of default keyfile
```

Comment:
Is this field required? Below it says ` std::vector<SNIEntry> _sniEntries;   // the first entry is the default server keyfile` ...

Reply:
Gone.

Reply:
_keyfile is still needed
_keyfileContent is actually gone now.